### PR TITLE
Added support for serverless.json. Improved error messages with filename

### LINKED ISSUE #3566: Support for Serverless.JSON
# This is a Feature Proposal
JSON support as Serverless config file (Serverless.json instead of Serverless.YML)
## Description

* Using YML feels awkard , most of the usage of lambda are done with NodeJS and Node mostly uses JSON file for configuration (Webpack, Grunt , package.json etc....)
Using YML doesn't feel right
* Most examples provided by AWS for cloudformation are using JSON ( examples here 
[AWS Labs](https://github.com/awslabs)

Similar or dependent issues:

This feature was already proposed but got lost in the different release [Optionnal Serverless.JSON](https://github.com/serverless/serverless/issues/1900)

### PR DESCRIPTION
**Pull request by HyperBrain**
## What did you implement:

Closes #3566 

## How did you implement it:

Service.load() now tries to find one of the supported service filename variants (`serverless.yaml`, `serverless.yml` and **new** `serverless.json`, with this precedence). As soon as one is found, it will be loaded. Additionally the filename is remembered and printed in the error messages. Previously `some error .... in serverless.yml` was printed regardless, if the file was named .yaml or .yml.

JSON service file support is added with no tech debt, as the used YAML parser is spec compliant (a JSON is itself a valid YAML per definition) and will load the JSON out-of-the-box.

## How can we verify it:

Convert any existing service.yml to JSON using some tool. Start your favorite serverless command.
It should behave and load exactly as before.

Additionally you can do some tests with multiple service files (.yml, .yaml, .json) to test the precedence - although I regard this as an edge case caused by sloppy development. ;-)

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/classes/Service.js [hunk 1]
```diff
@@ -33,23 +33,37 @@ class Service {
...
 
-    let serverlessYmlPath = path.join(servicePath, 'serverless.yml');
-    // change to serverless.yaml if the file could not be found
-    if (!this.serverless.utils.fileExistsSync(serverlessYmlPath)) {
-      serverlessYmlPath = path
-        .join(this.serverless.config.servicePath, 'serverless.yaml');
-    }
+    // List of supported service filename variants.
+    // The order defines the precedence.
+    const serviceFilenames = [
```
Comment (pmuens) [inline lib/classes/Service.js]: Unit tests which check for all the possible file extensions should be added

Comment (HyperBrain) [inline lib/classes/Service.js]: Yes, you're right. I will add them - also a precedence check would be beneficial.

Comment (pmuens) [inline lib/classes/Service.js]: Thanks 👍  +1 for the precedence check!

Comment (HyperBrain) [inline lib/classes/Service.js]: Tests added.

**Comment by HyperBrain** [comment]
Oh, I missed these. Thanks :)

**Comment by eahefnawy** [comment]
@HyperBrain sorry I didnt get a chance to test it earlier. Will move it to the next release to keep it in master for a while for "beta" testing. A bit worried about merging stuff during the release day! 😅
